silkworm/silkrpc/core/logs_walker.cpp

```diff
@@ -144,6 +144,10 @@ Task<void> LogsWalker::get_logs(std::uint64_t start, std::uint64_t end,
 
         if (!filtered_block_logs.empty()) {
             const auto block_with_hash = co_await core::read_block_by_number(block_cache_, *chain_storage, block_to_match);
+            if (!block_with_hash) {
+                SILK_ERROR << "LogsWalker::get_logs core::read_block_by_number block not found";
+                co_return;
+            }
```

Comment:
We need a more explicit error handling strategy here, instead.

Reply:
ok